lib/stagehand/auditor/checklist_visualizer.rb

```diff
@@ -0,0 +1,78 @@
+require 'graphviz'
+
+module Stagehand
+  module Auditor
+    class ChecklistVisualizer
+      def initialize(checklist)
+        entries = checklist.affected_entries.select(&:commit_id)
+
+        @graph = GraphViz.new( :G, :type => :graph );
+        @commits = Hash.new {|hash, commit_id| hash[commit_id] = Stagehand::Staging::CommitEntry.find(commit_id) }
+        @nodes = Hash.new
+        edges = Set.new
+
+        entries.group_by {|entry| commit_subject(entry) }.each do |subject, entries|
+          subgraph = create_subgraph(subject, @graph)
+          entries.each do |entry|
+            @nodes[entry] = create_node(entry, subgraph)
+          end
+        end
+
+        entries.group_by(&:key).each_value do |entries|
+          entries.combination(2).each do |entry_a, entry_b|
+            key = edge_key(entry_a, entry_b)
+            next if edges.include?(key)
+            next if @nodes[entry_a] == @nodes[entry_b]
+
+            edges << key
+            unless same_subject?(entry_a, entry_b)
+              create_edge(entry_a, entry_b, :color => :red, :fontcolor => :red, :label => edge_label(entry_a, entry_b))
+            end
+          end
+        end
```

Comment:
Can we split this code into a couple methods? It's a bit hard to read all in one method.